tests/ui_suite/fav_test.py

```diff
@@ -23,22 +32,33 @@ def test_add2fav(logined_page, fav_numbers: tuple, add_fav_from_detail: str):
                 fav_goods_names_expected.append(
                     LoginedPage.good_name_text(goods_for_fav[goods_index]))
                 # клик по сердцу - добавление в избранное
-                LoginedPage.goods_heart_button(
-                    goods_for_fav[goods_index]).click()
+                # with allure.step('Add desired goods to favorites using heart button'):
+                #     LoginedPage.goods_heart_button(goods_for_fav[goods_index]).click()
+                #     do_allure_attach(f'After clicking on the {goods_index}-th goods/s heart button')
+                LoginedPage.click_goods_heart_button(goods_for_fav, goods_index)
+
+
             case 'add_from_goods_detail':  # добавляем в избранное c товара
-                # проваливаемся в товар
-                goodsdetail_page = logined_page.goods_click(goods_index)
+                #with allure.step('Enter into goods details by clicking on the concrete goods'):
+                    # проваливаемся в товар
+                goodsdetail_page = logined_page.click_on_goods(goods_index)
                 # запоминаем товар
                 fav_goods_names_expected.append(goodsdetail_page.good_name_txt)
-                goodsdetail_page.fav_add_button.click()
-                goodsdetail_page.browser_back_button_click()
+                # with allure.step('Add current goods using heart button'):
+                #     goodsdetail_page.fav_add_button.click()
+                #     do_allure_screenshot(f'After clicking on the goods/s heart button on the goods detail poge')
```

Comment:
and this one, and then others below aswell

Reply:
done